libraries/rush-lib/src/logic/Git.ts

```diff
@@ -595,4 +595,23 @@ export class Git {
       throw e;
     }
   }
+  /**
+   *
+   * @param ref Given a ref which can be branch name, commit hash, tag name, etc, check if it is a commit hash
+   */
+  public isRefACommit(ref: string): boolean {
```

Comment:
To unit test this, you can pass in `HEAD` for an input that will always be valid, and the result of `git rev-parse HEAD` for a valid commit id.

Reply:
@dmichon-msft - Do you know of a single git command that will work for both of these scenarios (finding a merge base or confirming that something is a commit SHA)?

Reply:
Unfortunately not.

Reply:
Added some unit tests, please help to review if that's satisfactory enough 🙏